dev/bots/test.dart

```diff
@@ -602,7 +602,7 @@ Future<void> _runFlutterTest(String workingDirectory, {
   bq.TabledataResourceApi tableData,
   Map<String, String> environment,
 }) async {
-  final List<String> args = <String>['test']..addAll(options);
+  final List<String> args = <String>['test', ...options];
   if (flutterTestArgs != null && flutterTestArgs.isNotEmpty)
```

Comment:
Any chance of being able to catch this pattern?  The addAll inside this `if` could move into the literal as a spread. I think it would be clearer code, especially given that it's final.

Reply:
Yes, if it's a common enough pattern to make it worth the effort (and the potential UI clutter).

If we do support that pattern, then the question is whether it's better for users to have several small conversions that they need to repeatedly apply, so that they can pick and choose, or to have a single "convert everything" option that does as much as possible. I generally lean toward the former, but I'm not sure whether that's the right approach in this case.

Reply:
Not sure either, but single stepping seems more scalable to me (or at least, it doesn't seem like it needs to be a constraint to do it all at once).

In a quick poke over internal code, I see quite a few examples like this where the code looks like:

```
var list = [];
list..addAll(stuff ).... etc
```

instead of

```
var list = []..addAll()
```

do the quick fixes handle the former pattern as well?

Reply:
No. It could, but we'd have to ensure that `list` can't be altered prior to the invocation of `addAll`. What that probably amounts to is that the assist would likely handle a very small number of cases, at least initially.